yt/frontends/amrvac/data_structures.py

```diff
@@ -203,20 +203,8 @@ def __init__(
         # note: geometry_override and parfiles are specific to this frontend
 
         self._geometry_override = geometry_override
-        super().__init__(
-            filename,
-            dataset_type,
-            units_override=units_override,
-            unit_system=unit_system,
-            default_species_fields=default_species_fields,
-        )
+        self._parfiles = None
```

Comment:
using an empty list here allows some simplifications in the rest of your patch
```suggestion
        self._parfiles = []
```